cfgagent/vlanconf.cpp

```diff
@@ -0,0 +1,475 @@
+            string admin_status;
+            uint32_t mtu = 0;
+            vector<FieldValueTuple> fvVector;
+            string members;
+
+            /* Add host VLAN when it has not been created. */
+            if (m_vlans.find(key) == m_vlans.end())
+            {
+                addHostVlan(vlan_id);
+            }
+
+            /* set up host env .... */
+            for (auto i : kfvFieldsValues(t))
+            {
+                /* Set port admin status */
+                if (fvField(i) == "admin_status") {
+                    admin_status = fvValue(i);
+                    setHostVlanAdminState(vlan_id, admin_status);
+                    fvVector.push_back(i);
+                }
+                /* Set port mtu */
+                else if (fvField(i) == "mtu") {
+                    mtu = (uint32_t)stoul(fvValue(i));
+                    setHostVlanMtu(vlan_id, mtu);
+                    fvVector.push_back(i);
+                }
+                else if (fvField(i) == "members@") {
+                    members = fvValue(i);
+                }
+            }
+            /* fvVector should not be empty */
+            if (fvVector.empty())
+            {
+                FieldValueTuple a("admin_status",  "up");
+                fvVector.push_back(a);
+            }
+            m_appVlanTableProducer.set(key, fvVector);
+            m_vlans.insert(key);
+
+            fvVector.clear();
+            FieldValueTuple s("state", "ok");
+            fvVector.push_back(s);
+            m_stateVlanTable.set(key, fvVector);
+
+            it = consumer.m_toSync.erase(it);
+
+            /*
+             * Members configured together with VLAN in untagged mode.
+             * This is to be compatible with access VLAN configuration from minigraph.
+             */
+            if (!members.empty())
+            {
+                processUntaggedVlanMembers(key, members);
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (m_vlans.find(key) != m_vlans.end())
+            {
+                removeHostVlan(vlan_id);
+                m_vlans.erase(key);
+                m_appVlanTableProducer.del(key);
+                m_stateVlanTable.del(key);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("%s doesn't exist", key.c_str());
+            }
+            SWSS_LOG_DEBUG("%s", (dumpTuple(consumer, t)).c_str());
+            it = consumer.m_toSync.erase(it);
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+            SWSS_LOG_DEBUG("%s", (dumpTuple(consumer, t)).c_str());
+            it = consumer.m_toSync.erase(it);
+        }
+    }
+}
+
+bool VlanConf::isMemberStateOk(string &alias)
+{
+    vector<FieldValueTuple> temp;
+
+    if (!alias.compare(0, strlen(LAG_PREFIX), LAG_PREFIX))
+    {
+        if (m_stateLagTable.get(alias, temp))
+        {
+            SWSS_LOG_DEBUG("%s is ready\n", alias.c_str());
```

Comment:
>\n [](start = 39, length = 2)

remove this '\n', does dot look like we need it.

Reply:
ok